### PR TITLE
Adding url path parsing depending on path items

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding url path based on whether path items are present and fixing err variable assignment on body params.

### CODE REVIEW

File: test/storage/2019-07-07/azblob/service.go [hunk 1]
```diff
@@ -57,7 +56,7 @@ func (client *serviceOperations) GetAccountInfo(ctx context.Context) (*ServiceGe
 }
 
 // getAccountInfoCreateRequest creates the GetAccountInfo request.
-func (client *serviceOperations) getAccountInfoCreateRequest(urlParameter string) (*azcore.Request, error) {
+func (client *serviceOperations) getAccountInfoCreateRequest() (*azcore.Request, error) {
 	urlPath := "/"
```
Comment (jhendrixMSFT) [inline test/storage/2019-07-07/azblob/service.go]: Would be good to remove this, avoiding the call to `client.u.Parse()` thus removing creating a new `url.URL` (I'm fine with deferring it to a separate PR if that works better).

File: src/generator/helpers.ts [hunk 2]
```diff
@@ -164,13 +167,17 @@ export function generateParameterInfo(op: Operation): ParamInfo[] {
     }
     // include client and method params
     const global = param.implementation === ImplementationLocation.Client;
-    params.push(new paramInfo(param.language.go!.name, param.schema.language.go!.name, global, param.required === true));
+    let isHost = false;
+    if (global) {
+      isHost = (param.extensions!['x-ms-priority'] === 0 && param.extensions!['x-in'] === 'path');
```
Comment (jhendrixMSFT) [inline src/generator/helpers.ts]: Is the `!` going to be problematic?  `extensions` can legitimately be undefined.

Comment (catalinaperalta) [reply src/generator/helpers.ts]: I can add an if to check if it's undefined before

Comment (catalinaperalta) [reply src/generator/helpers.ts]: if (global && param.extensions !== undefined)

Comment (jhendrixMSFT) [reply src/generator/helpers.ts]: If you use `?` instead it gives you the same effect.

Comment (catalinaperalta) [reply src/generator/helpers.ts]: ok 👍

**Pull request merged**